packages/utils/src/lib/report-to-md.ts

```diff
@@ -25,107 +51,246 @@ function reportToHeaderSection(): string {
+  plugins: ScoredReport['plugins'],
+): string {
+  const plugin = plugins.find(({ slug }) => slug === refPlugin) as PluginReport;
+  const group = plugin?.groups?.find(({ slug }) => slug === refSlug);
+  const groupScore = Number(formatReportScore(group?.score || 0));
+
+  if (!group) {
+    throwIsNotPresentError(`Group ${refSlug}`, plugin?.slug);
+  }
+
+  const groupTitle = li(
+    `${getRoundScoreMarker(groupScore)} ${group.title} (_${plugin.title}_)`,
+  );
+  const foundAudits = group.refs.reduce((acc, ref) => {
+    const audit = plugin?.audits.find(
+      ({ slug: auditSlugInPluginAudits }) =>
+        auditSlugInPluginAudits === ref.slug,
+    );
+    if (audit) {
+      acc.push(audit);
+    }
+
+    return acc;
+  }, [] as AuditReport[]);
+
+  const groupAudits = foundAudits.reduce((acc, audit) => {
+    const auditTitle = link(`#${audit.slug}`, audit?.title);
+    acc += `  ${li(
+      `${getSquaredScoreMarker(audit.score)} ${auditTitle} - **${getAuditResult(
+        audit,
+      )}**`,
+    )}`;
+    acc += NEW_LINE;
+    acc += NEW_LINE;
+    return acc;
+  }, '');
+
+  return groupTitle + NEW_LINE + NEW_LINE + groupAudits + NEW_LINE + NEW_LINE;
 }
 
-function reportToDetailSection(report: ScoredReport): string {
-  let md = '';
-  const { categories, plugins } = report;
+function reportToAuditsSection(report: ScoredReport): string {
+  const auditsData = report.plugins.reduce((acc, plugin) => {
+    const audits = plugin.audits.reduce((acc, audit) => {
+      const auditTitle = `<a name="${audit.slug}"></a>${audit.title} (${plugin.title})`;
+      const detailsTitle = `${getSquaredScoreMarker(
+        audit.score,
+      )} ${getAuditResult(audit)} (score: ${formatReportScore(audit.score)})`;
 
-  categories.forEach(category => {
-    const { title, refs, score } = category;
+      acc += h3(auditTitle);
+      acc += NEW_LINE;
+      acc += NEW_LINE;
 
-    md += style(`${title} ${score}`) + NEW_LINE;
+      if (!audit.details?.issues?.length) {
+        acc += details(detailsTitle, 'No details');
+        acc += NEW_LINE;
+        acc += NEW_LINE;
+        return acc;
+      }
 
-    md +=
-      refs
-        .map(
-          ({ slug: auditSlugInCategoryRefs, weight, plugin: pluginSlug }) => {
-            const plugin = plugins.find(({ slug }) => slug === pluginSlug);
-
-            if (!plugin) {
-              // this should never happen
-              throw new Error(
-                `plugin ${pluginSlug} not present in config.plugins`,
-              );
-            }
+      const detailsTable = [
+        detailsTableHeaders,
+        ...audit.details.issues.map(issue => [
+          `${getSeverityIcon(issue.severity)} <i>${issue.severity}</i>`,
+          issue.message,
+          `<a href="${toUnixPath(issue.source?.file as string, {
+            toRelative: true,
+          })}">
+             <code>${issue.source?.file}</code>
+           </a>`,
+          String(issue.source?.position?.startLine),
```

Comment:
Added fallback for string.
For the line range, added a functionality where with no `startLine` it's blank, without `endLine` or with `startLine === endLine` only startLine is shown, otherwise it's shown in a format `startLine + '-' + endLine`, for example `7-11`.